satpy/readers/abi_l1b.py

```diff
@@ -139,13 +147,27 @@ def _vis_calibrate(self, data):
         res.attrs['standard_name'] = 'toa_bidirectional_reflectance'
         return res
 
+    def _get_minimum_radiance(self, data):
+        """Estimate minimum radiance from Rad DataArray."""
+        attrs = data.attrs
+        scale_factor = attrs["scale_factor"]
+        add_offset = attrs["add_offset"]
+        count_zero_rad = - add_offset / scale_factor
+        count_pos = np.ceil(count_zero_rad)
+        min_rad = count_pos * scale_factor + add_offset
+        return min_rad
+
     def _ir_calibrate(self, data):
         """Calibrate IR channels to BT."""
         fk1 = float(self["planck_fk1"])
         fk2 = float(self["planck_fk2"])
         bc1 = float(self["planck_bc1"])
         bc2 = float(self["planck_bc2"])
 
+        if self.clip_negative_radiances:
+            min_rad = self._get_minimum_radiance(data)
```

Comment:
Do I understand correctly that data is not clean radiances and that 0 radiance isn't 0 in data? the above `_rad_calibrate` function seams to indicate the opposite...

Reply:
@mraspaud I don't understand your question ... can you try to reformulate it ?

Reply:
Ok, so I though the data in the file was radiances, so I'm surprise you can't just do `data.clip(min=0)`. The `_rad_calibrate` method tells me that no scale and offset is necessary to convert the data in the file in any way, it looks like it's already radiances...

Reply:
You can't clip to 0 because then in the following line (`res = (fk2 / np.log(fk1 / data + 1) - bc1) / bc2`),  in `np.log(fk1 / data + 1)` np.nan would pop up by if data == 0. So we need to specify values close 0 like 0.001 ... but using 0.001 or 0.0001 can change brightness temperature by more 10 K. We therefore infer the minimum allowed radiance from the scale/offsets ...

Reply:
Side note: The values in the file on-disk are scaled radiances (need scale factor and offset applied). These are loadable as calibration="counts".

Reply:
Ok, that make sense, thanks for clarifying.